unit-tests/src/org/commcare/CommCareTestApplication.java

```diff
@@ -214,4 +222,25 @@ public ModernHttpRequester buildModernHttpRequester(Context context, URL url,
     public DataPullRequester getDataPullRequester() {
         return LocalDataPullResponseFactory.INSTANCE;
     }
+
+    @Override
+    public void afterTest(Method method) {
+        Robolectric.flushBackgroundThreadScheduler();
+        if (asyncExceptions.size() > 0) {
+            for(Throwable throwable: asyncExceptions) {
+                throwable.printStackTrace();
+                fail("Test failed with " + asyncExceptions.size() + " asyncronous exceptions.");
```

Comment:
Ok, in that case can you just change the message to be more clear? IE:

"Test failed due to [2] threads crashing off the main thread. See error log for more details"